brainglobe_stitch/tile.py

```diff
@@ -55,3 +55,148 @@ def __init__(
+
+
+class Overlap:
+    """
+    Overlap class to store information about the overlap between two tiles.
+
+    Attributes
+    ----------
+    coordinates : npt.NDArray
+        The coordinates for the start of the overlap in the fused image.
+    size : List[npt.NDArray]
+        The size of the overlap in the fused image for each resolution pyramid.
+    tiles : Tuple[Tile, Tile]
+        The overlapping tiles.
+    local_coordinates : List[Tuple[npt.NDArray, npt.NDArray]]
+        A list of tuples representing the starting coordinates for the overlap
+        in each tile's data pyramid for each resolution pyramid level.
+        The tuple contains the starting coordinates for the overlap in each
+        tile.
+    """
+
+    def __init__(
+        self,
+        overlap_coordinates: npt.NDArray,
+        overlap_size: npt.NDArray,
+        tile_i: Tile,
+        tile_j: Tile,
+    ):
+        """
+        Initialize the overlap.
+
+        Parameters
+        ----------
+        overlap_coordinates: npt.NDArray
+            The starting coordinates for the overlap in the fused image.
+        overlap_size: npt.NDArray
+            The size of the overlap in the fused image [z,y,x].
+        tile_i: Tile
+            The first tile.
+        tile_j: Tile
+            The second tile.
+        """
+        self.coordinates: npt.NDArray = overlap_coordinates
+        self.size: List[npt.NDArray] = [overlap_size]
+        self.tiles: Tuple[Tile, Tile] = (tile_i, tile_j)
+        self.local_coordinates: List[Tuple[npt.NDArray, npt.NDArray]] = [
+            (
+                np.zeros(self.coordinates.shape),
+                np.zeros(self.coordinates.shape),
+            ),
+        ]
+        self.get_local_overlap_coordinates()
+
+    def get_local_overlap_coordinates(self) -> None:
+        """
+        Calculate the starting coordinates for the overlap in
+        each tile's data pyramid.
+        """
+        tile_shape = self.tiles[0].data_pyramid[0].shape
+        resolution_pyramid = self.tiles[0].resolution_pyramid
+
+        for i in range(self.coordinates.shape[0]):
+            if self.tiles[0].position[i] < self.tiles[1].position[i]:
+                self.local_coordinates[0][0][i] = (
+                    tile_shape[i] - self.size[0][i]
+                )
+                self.local_coordinates[0][1][i] = 0
+            else:
+                self.local_coordinates[0][0][i] = 0
+                self.local_coordinates[0][1][i] = (
+                    tile_shape[i] - self.size[0][i]
+                )
+
+        for j in range(1, len(resolution_pyramid)):
+            self.local_coordinates.append(
+                (
+                    self.local_coordinates[0][0] // resolution_pyramid[j],
+                    self.local_coordinates[0][1] // resolution_pyramid[j],
+                )
+            )
+            self.size.append(self.size[0] // resolution_pyramid[j])
+
+    def extract_tile_overlaps(
```

Comment:
Do you think that these two methods will be useful for the plugin as well (and not uniquely to testing)?

Reply:
They'll be used in upcoming PRs for replacing overlapping areas after interpolating the stitched area.